tests/integration/explain.go

```diff
@@ -97,41 +101,40 @@ type ExplainRequest struct {
 }
 
 func executeExplainRequest(
-	ctx context.Context,
-	t *testing.T,
-	description string,
-	db client.DB,
+	s *state,
 	action ExplainRequest,
 ) {
 	// Must have a non-empty request.
 	if action.Request == "" {
-		require.Fail(t, "Explain test must have a non-empty request.", description)
+		require.Fail(s.t, "Explain test must have a non-empty request.", s.testCase.Description)
 	}
 
 	// If no expected results are provided, then it's invalid use of this explain testing setup.
 	if action.ExpectedError == "" &&
 		action.ExpectedPatterns == nil &&
 		action.ExpectedTargets == nil &&
 		action.ExpectedFullGraph == nil {
-		require.Fail(t, "Atleast one expected explain parameter must be provided.", description)
+		require.Fail(s.t, "Atleast one expected explain parameter must be provided.", s.testCase.Description)
 	}
 
 	// If we expect an error, then all other expected results should be empty (they shouldn't be provided).
 	if action.ExpectedError != "" &&
 		(action.ExpectedFullGraph != nil ||
 			action.ExpectedPatterns != nil ||
 			action.ExpectedTargets != nil) {
-		require.Fail(t, "Expected error should not have other expected results with it.", description)
+		require.Fail(s.t, "Expected error should not have other expected results with it.", s.testCase.Description)
 	}
 
-	result := db.ExecRequest(ctx, action.Request)
-	assertExplainRequestResults(
-		ctx,
-		t,
-		description,
-		&result.GQL,
-		action,
-	)
+	for _, node := range getNodes(action.NodeID, s.nodes) {
```

Comment:
question: Can you write a line here to document what the addition of this for loop enables?

Reply:
This loop is a copy-paste of the same line in the other 20 or so actions.  I don't really think it needs to chew up anymore eyespace.  It is not unusual.